controllers/helmrepository_controller_oci.go

```diff
@@ -366,3 +387,42 @@ func (r *HelmRepositoryOCIReconciler) eventLogf(ctx context.Context, obj runtime
 	}
 	r.Eventf(obj, eventType, reason, msg)
 }
+
+// oidcAuth generates the OIDC credential authenticator based on the specified cloud provider.
+func oidcAuth(ctx context.Context, obj *sourcev1.HelmRepository) (helmreg.LoginOption, error) {
+	url := strings.TrimPrefix(obj.Spec.URL, helmreg.OCIScheme)
```

Comment:
This trim results in URL strings like `://fluxtestallowedochimp.azurecr.io/mydemo`.
The `name.ParseReference()` below fails to parse it properly, resulting in `ref.Context().RegistryStr()` in the oci/auth package to return empty string.

Doing
```suggestion
	url := strings.TrimPrefix(obj.Spec.URL, helmreg.OCIScheme+"://")
```
makes it work.
It worked fine in case of aws because that uses regex pattern match on the image string, not using `ref`.

Reply:
thanks for testing 👍